s3/session.go

```diff
@@ -47,13 +48,35 @@ func findBucketRegion(bucket string, config *aws.Config) (string, error) {
 }
 
 // TODO : unit tests
-func getAWSRegion(s3Bucket string, config *aws.Config, settings map[string]string) (string, error) {
+func GetAWSRegion(s3Bucket string, config *aws.Config, settings map[string]string) (string, error) {
 	if region, ok := settings[RegionSetting]; ok {
 		return region, nil
 	}
+
 	if config.Endpoint == nil ||
-		*config.Endpoint == "" ||
-		strings.HasSuffix(*config.Endpoint, ".amazonaws.com") {
+		*config.Endpoint == "" {
+
+		if config.Endpoint != nil {
+			hostAddr, parseErr := url.Parse(*config.Endpoint)
+
+			if parseErr != nil {
+				return "us-east-1", parseErr
+			}
+
+			host, _, err := net.SplitHostPort(hostAddr.Host)
+
+			if err != nil {
```

Comment:
In this logic, if WAL-G failed to parse the endpoint host, it won't attempt to call `findBucketRegion`, is it intended behavior?
I think that WAL-G should always try both approaches before giving up.